craft_application/errors.py

```diff
@@ -120,6 +120,26 @@ def __init__(self, platform: str, all_platforms: Sequence[str]) -> None:
         super().__init__(message=message, details=details)
 
 
+class EmptyBuildPlanError(CraftError):
+    """The build plan filtered out all possible builds."""
+
+    def __init__(self) -> None:
+        message = "No build matches the current platform."
+        resolution = 'Check the "--platform" and "--build-for" parameters.'
```

Comment:
`--build-for` as a CLI option is Snapcraft specific

Reply:
Then this is something we need to fix, because [it's enabled everywhere](https://github.com/canonical/craft-application/blob/main/craft_application/commands/lifecycle.py#L130)
In the meantime, do you have a suggestion for the message/resolution?